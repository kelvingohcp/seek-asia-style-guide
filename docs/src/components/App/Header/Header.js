import styles from './Header.less';
import brandStyle from './Header.css.js';

import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';
import { PageBlock, Card, Section, Text, StyleGuideContext, Constants } from 'seek-asia-style-guide/react';
import Logo from './Logo/Logo';
import demoSpecExports from '../../../../../react/*/*.demo.js';
import jobStreetDemoExports from '../../../../../jobStreet/*/*.demo.js';
import jobsDBDemoExports from '../../../../../jobsDB/*/*.demo.js';
const skDemoSpecs = demoSpecExports
  .map(x => x.default)
  .filter(sk => sk.seekComponent === true);
const saDemoSpecs = demoSpecExports
  .concat(jobStreetDemoExports, jobsDBDemoExports)
  .map(x => x.default)
  .filter(function(el) {
    return skDemoSpecs.indexOf(el) < 0;
  });

const { SEEKASIA, JOBSDB, JOBSTREET } = Constants;

export default class Header extends Component {
  constructor(props) {
    super(props);

    this.state = {
      menuOpen: false
    };
  }

  handleMenuToggle = event => {
    this.setState({ menuOpen: event.target.checked });
  };

  handleMenuClose = () => {
    this.setState({ menuOpen: false });
  };

  render() {
    const { changeTenant } = this.props;
    const { menuOpen } = this.state;

    const headerClasses = classnames({
      [brandStyle.headerBlock]: true,
      [styles.fixedHeaderBlock]: menuOpen
    });

    return (
      <div>
        <PageBlock className={headerClasses}>
          <Section className={styles.headerSection}>
            <div className={styles.sectionContent}>
              <Link
                className={classnames([styles.logoLink, brandStyle.logoLink])}
                to='/'
                onClick={this.handleMenuClose} >
                <StyleGuideContext.Consumer>
                  {({ tenant }) => (
                    <Logo
                      svgClassName={classnames([styles.logo, brandStyle.logo])}
                      tenant={tenant}
                    />
                  )}
                </StyleGuideContext.Consumer>
              </Link>

              <div className={styles.hamburger}>
                <input
                  onChange={this.handleMenuToggle}
                  checked={menuOpen}
                  id='hamburgerCheckbox'
                  type='checkbox'
                  className={styles.checkbox}
                />
                <label htmlFor='hamburgerCheckbox' className={styles.bars}>
                  <div className={classnames([styles.bar1, brandStyle.bar])} />
                  <div className={classnames([styles.bar2, brandStyle.bar])} />
                  <div className={classnames([styles.bar3, brandStyle.bar])} />
                  <span className={styles.hamburgerText}>Show menu</span>
                </label>

                <div className={styles.menu} onClick={this.handleMenuClose}>
                  <PageBlock>
                    <Section header>
                      <Card transparent className={styles.components}>
                        <h2>
                          <Text yelling regular>
                            <Link className={styles.link} to='./'>
                              Home
                            </Link>
                          </Text>
                        </h2>
                      </Card>

                      <Card transparent>
                        <h2>
                          <Text
                            yelling
                            className={styles.componentSectionTitle}>
                            Tenants
                          </Text>
                        </h2>
                      </Card>
                      <Card transparent className={styles.componentList}>
                        <Text
                          shouting
                          regular
                          className={styles.componentListItem}>
                          <Link className={styles.link} to='#' onClick={() => changeTenant(SEEKASIA)}>
                            SeekAsia
                          </Link>
                        </Text>
                        <Text
                          shouting
                          regular
                          className={styles.componentListItem}>
                          <Link className={styles.link} to='#' onClick={() => changeTenant(JOBSDB)}>
                            JobsDB
                          </Link>
                        </Text>
                        <Text
                          shouting
                          regular
                          className={styles.componentListItem}>
                          <Link className={styles.link} to='#' onClick={() => changeTenant(JOBSTREET)}>
                            JobStreet
                          </Link>
                        </Text>
                      </Card>

                      <Card transparent>
                        <h2>
                          <Text
                            yelling
                            className={styles.componentSectionTitle}>
                            SEEK Asia Style Basics
                          </Text>
                        </h2>
                      </Card>
                      <Card transparent className={styles.componentList}>
                        <Text
                          shouting
                          regular
                          className={styles.componentListItem}>
                          <Link className={styles.link} to='./colors'>
                            Colors
                          </Link>
                        </Text>
                        <Text
                          shouting
                          regular
                          className={styles.componentListItem}>
                          <Link className={styles.link} to='./typography'>
                            Typography
                          </Link>
                        </Text>
                      </Card>

                      <Card transparent>
                        <h2>
                          <Text
                            yelling
                            className={styles.componentSectionTitle}>
                            Tools
                          </Text>
                        </h2>
                      </Card>
                      <Card transparent className={styles.componentList}>
                        <Text
                          shouting
                          regular
                          className={styles.componentListItem}>
                          <Link className={styles.link} to='./sandbox'>
                            Sandbox
                          </Link>
                        </Text>
                      </Card>

                      <Card transparent>
                        <h2>
                          <Text
                            yelling
                            className={styles.componentSectionTitle}>
                            SEEK Asia Components
                          </Text>
                        </h2>
                      </Card>
                      <Card transparent className={styles.componentList}>
                        {saDemoSpecs.map(demoSpec => (
                          <Text
                            shouting
                            regular
                            key={demoSpec.title}
                            className={styles.componentListItem}>
                            <Link className={styles.link} to={demoSpec.route}>
                              {demoSpec.title}
                            </Link>
                          </Text>
                        ))}
                      </Card>

                      <Card transparent>
                        <h2>
                          <Text
                            yelling
                            className={styles.componentSectionTitle}>
                            SEEK Components
                          </Text>
                        </h2>
                      </Card>
                      <Card transparent className={styles.componentList}>
                        {skDemoSpecs.map(demoSpec => (
                          <Text
                            shouting
                            regular
                            key={demoSpec.title}
                            className={styles.componentListItem}>
                            <Link className={styles.link} to={demoSpec.route}>
                              {demoSpec.title}
                            </Link>
                          </Text>
                        ))}
                      </Card>
                    </Section>
                  </PageBlock>
                </div>
              </div>
            </div>
          </Section>
        </PageBlock>
      </div>
    );
  }
}

Header.propTypes = {
  changeTenant: PropTypes.func
};
