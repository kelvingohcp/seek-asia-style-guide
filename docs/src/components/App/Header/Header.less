@import (reference) "~seek-asia-style-guide/theme";

.headerBlock {
  background: @sk-blue;
}

.fixedHeaderBlock {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
}

.logoLink {
  display: inline-block;
}

.logo {
  width: 118px;
  height: 31px;
  position: relative;
  top: 3px;
  margin-bottom: 3px;
}

.title {
  .yellingTextResponsive();

  text-transform: lowercase;
  position: absolute;
  color: @sk-white;
  text-decoration: none;
  top: -2px;
  left: 126px;
  width: 150px;
}

.sectionContent {
  display: flex;
  align-items: center;
  justify-content: space-between;
  min-height: @grid-base * 10;
}

.hamburgerText {
  overflow: hidden;
  width: 0;
  height: 0;
  text-indent: 999px;
  position: absolute;
}

.checkbox {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.bars {
  display: block;
  cursor: pointer;
  width: @grid-gutter-width * 2;
  height: @grid-row-height * 7;
  position: relative;
  transform-origin: 100% 50%;
  transform: scale(0.7);
}

@barHeight: 6px;
.bar() {
  height: @barHeight;
  width: 100%;
  background: @sk-white;
  position: absolute;
  transition: all 0.1s ease;
}

.bar1 {
  .bar();
  top: 0;
  .checkbox:checked ~ .bars & {
    transform: translateY(11px) rotate(45deg);
  }
}

.bar2 {
  .bar();
  top: 50%;
  margin-top: @barHeight * -0.5;
  .checkbox:checked ~ .bars & {
    opacity: 0;
  }
}

.bar3 {
  .bar();
  bottom: 0;
  .checkbox:checked ~ .bars & {
    transform: translateY(-11px) rotate(-45deg);
  }
}

.menu {
  background: @sk-blue;
  color: @sk-white;
  position: fixed;
  z-index: 10;
  top: @grid-row-height * 16;
  left: 0;
  right: 0;
  bottom: 0;
  padding-bottom: @grid-row-height * 20;
  overflow: auto;
  display: none;

  .checkbox:checked ~ & {
    display: block;
  }
}

.componentList {
  display: flex;
  flex-wrap: wrap;
}

.componentListItem {
  flex: 0 0 (@grid-base * 64);
}

.componentSectionTitle {
  padding-top: @grid-base * 8;
  border-bottom: medium solid rgba(255, 255, 255, 0.2);
}

.link {
  color: white;
  text-decoration: none;
  display: inline-block;
  margin-bottom: @grid-row-height * 2;
  &:hover {
    box-shadow: inset 0 -2px 0 @sk-white;
  }
}
