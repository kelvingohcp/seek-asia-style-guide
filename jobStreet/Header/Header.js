import React from 'react';
import PropTypes from 'prop-types';
import styles from './Header.less';
import { Header as GlobalHeader } from 'seek-asia-style-guide/react';
import Logo from '../Logo/Logo';
import { Icon, MoreIcon } from 'seek-asia-style-guide/react';
import { getLocalization, locales } from '../localization';

const getJobStreetProps = ({ country, language }) => {
  const messages = getLocalization({ country, language });

  const more = [
    { title: messages['header.overseasJobsTitle'], url: messages['header.overseasJobsUrl'] },
    { title: messages['header.freshGradJobsTitle'], url: messages['header.freshGradJobsUrl'] },
    { title: messages['header.classifiedJobsTitle'], url: messages['header.classifiedJobsUrl'] }
  ];

  const links = [
    { title: messages['header.homeTitle'], url: messages['header.homeUrl'], ItemIcon: <Icon type="home" size="small" /> },
    { title: messages['header.myJobStreetTitle'], url: messages['header.myJobStreetUrl'], ItemIcon: <Icon type="myJobsDB" size="small" /> },
    { title: messages['header.companyProfilesTitle'], url: messages['header.companyProfilesUrl'], ItemIcon: <Icon type="company" size="small" /> },
    { title: messages['header.careerInsightsTitle'], url: messages['header.careerInsightsUrl'], ItemIcon: <Icon type="careerInsight" size="small" /> },
    { title: messages['header.educationTitle'], url: messages['header.educationUrl'], ItemIcon: <Icon type="education" size="small" /> },
    { title: messages['menu.more'], url: '', ItemIcon: MoreIcon, children: more, hideInDesktop: true }
  ];

  return {
    links,
    messages,
    locales
  };
};

const Header = ({ country = 'my', language = 'en', activeTab, loginAvailable = false, ...restProps }) => {
  return (
    <GlobalHeader
      LogoComponent={Logo}
      logoProps={{ country }}
      activeTab={activeTab}
      loginAvailable={loginAvailable}
      {...getJobStreetProps({ country, language })}
      brandStyles={styles}
      locales={locales}
      country={country}
      language={language}
      {...restProps}
    />
  );
};

Header.propTypes = {
  country: PropTypes.string,
  language: PropTypes.string,
  activeTab: PropTypes.string,
  loginAvailable: PropTypes.bool
};

export default Header;
