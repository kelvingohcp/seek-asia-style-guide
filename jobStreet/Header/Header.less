@import (reference) "~seek-asia-style-guide/theme";

.primaryNavLink,
.primaryNavLink:visited {
  color: @jsBlue;
}

.primaryNavLink:hover {
  color: darken(@jsBlue, 10%);
}

.activeActionTrayIcon {
  fill: @jsBlue;
}

.menuIcon {
  color: @jsBlue;
  fill: currentColor;
}

