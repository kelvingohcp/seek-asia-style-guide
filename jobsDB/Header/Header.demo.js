import React from 'react';
import PropTypes from 'prop-types';
import { Header } from 'seek-asia-style-guide/jobsDB';
import { Constants } from 'seek-asia-style-guide/react';
import { Link } from 'react-router-dom';
import { AUTHENTICATED, UNAUTHENTICATED } from 'seek-asia-style-guide/react/private/authStatusTypes';

export const makeDummyLinkRendererForPath = path => {
  const DummyLinkRenderer = ({ href, ...props }) => (
    <Link to={`${path}#actualHref=${href}`} {...props} />
  );

  DummyLinkRenderer.propTypes = {
    href: PropTypes.string
  };

  return DummyLinkRenderer;
};

const { ACTIVE_TAB_HOME, ACTIVE_TAB_SEARCH, ACTIVE_TAB_SAVED_JOBS } = Constants;
const ROUTE = '/jobsdb-header';
const domain = 'jobsdb.com';

export default {
  route: ROUTE,
  tenantPath: 'jobsDB',
  title: 'JobsDB Header',
  component: Header,
  initialProps: {
    language: 'en',
    country: 'hk',
    loginAvailable: true,
    activeTab: ACTIVE_TAB_HOME,
    selectCountry: true,
    domainUrl: domain,
    linkRenderer: makeDummyLinkRendererForPath(ROUTE),
    authenticationStatus: UNAUTHENTICATED,
    showCountryLanguage: true
  },
  options: [{
    label: 'States',
    type: 'checklist',
    states: [
      {
        label: 'Hide Tray',
        transformProps: props => ({
          ...props,
          actionTrayProps: {
            ...props.actionTrayProps,
            showTray: false
          }
        })
      }, {
        label: 'Hide Home',
        transformProps: props => ({
          ...props,
          actionTrayProps: {
            ...props.actionTrayProps,
            showHome: false
          }
        })
      }, {
        label: 'Hide Country',
        transformProps: props => ({
          ...props,
          showCountryLanguage: false
        })
      }, {
        label: 'Hide Search',
        transformProps: props => ({
          ...props,
          actionTrayProps: {
            ...props.actionTrayProps,
            showSearch: false
          }
        })
      }, {
        label: 'Hide Saved Jobs',
        transformProps: props => ({
          ...props,
          actionTrayProps: {
            ...props.actionTrayProps
          }
        })
      }, {
        label: 'Hide Menu',
        transformProps: props => ({
          ...props,
          actionTrayProps: {
            ...props.actionTrayProps,
            showMenu: false
          }
        })
      },
      {
        label: 'Hide Navigation',
        transformProps: props => ({
          ...props,
          hideNavMenu: true
        })
      }
    ]
  }, {
    label: 'Locale',
    type: 'radio',
    states: [
      {
        label: 'Hong Kong (English)',
        transformProps: props => ({
          ...props,
          language: 'en',
          country: 'hk',
          baseUrl: `https://hk.${domain}`
        })
      },
      {
        label: 'Indonesia (English)',
        transformProps: props => ({
          ...props,
          language: 'en',
          country: 'id',
          baseUrl: `https://id.${domain}`
        })
      },
      {
        label: 'Indonesia (Bahasa)',
        transformProps: props => ({
          ...props,
          language: 'id',
          country: 'id',
          baseUrl: `https://id.${domain}`
        })
      },
      {
        label: 'Singapore (English)',
        transformProps: props => ({
          ...props,
          language: 'en',
          country: 'sg',
          baseUrl: `https://sg.${domain}`
        })
      },
      {
        label: 'Thailand (English)',
        transformProps: props => ({
          ...props,
          language: 'en',
          country: 'th',
          baseUrl: `https://th.${domain}`
        })
      },
      {
        label: 'Thailand (Thai)',
        transformProps: props => ({
          ...props,
          language: 'th',
          country: 'th',
          baseUrl: `https://th.${domain}`
        })
      }
    ]
  }, {
    label: 'Login Available',
    type: 'radio',
    states: [
      {
        label: 'Login available',
        transformProps: props => ({
          ...props,
          loginAvailable: true
        })
      },
      {
        label: 'Login unavailable',
        transformProps: props => ({
          ...props,
          loginAvailable: false
        })
      }
    ]
  }, {
    label: 'Logged In',
    type: 'radio',
    states: [
      {
        label: 'UnAuthenticated',
        transformProps: props => ({
          ...props,
          authenticationStatus: UNAUTHENTICATED,
          loginAvailable: true
        })
      },
      {
        label: 'Authenticated',
        transformProps: props => ({
          ...props,
          authenticationStatus: AUTHENTICATED,
          loginAvailable: true,
          userName: 'Olivia'
        })
      },
      {
        label: 'Pending',
        transformProps: props => ({
          ...props,
          returnUrl: '/jobs',
          loginAvailable: false
        })
      }
    ]
  }, {
    label: 'Select Country',
    type: 'radio',
    states: [
      {
        label: 'Select Country',
        transformProps: props => ({
          ...props,
          selectCountry: true
        })
      },
      {
        label: 'Not Select Country',
        transformProps: props => ({
          ...props,
          selectCountry: false
        })
      }
    ]
  }, {
    label: 'Active tab',
    type: 'radio',
    states: [
      {
        label: 'Home',
        transformProps: props => ({
          ...props,
          activeTab: ACTIVE_TAB_HOME
        })
      },
      {
        label: 'Search',
        transformProps: props => ({
          ...props,
          activeTab: ACTIVE_TAB_SEARCH
        })
      },
      {
        label: 'Saved Jobs',
        transformProps: props => ({
          ...props,
          activeTab: ACTIVE_TAB_SAVED_JOBS
        })
      }
    ]
  }]
};
