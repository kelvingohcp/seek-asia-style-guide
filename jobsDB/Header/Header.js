import React from 'react';
import PropTypes from 'prop-types';
import styles from './Header.less';
import { Header as GlobalHeader } from 'seek-asia-style-guide/react';
import Logo from '../Logo/Logo';
import { JobFunctionIcon, ProfileIcon, JobInvitationIcon, Icon } from 'seek-asia-style-guide/react';
import { getLocalization } from '../localization';
import { AUTHENTICATED, UNAUTHENTICATED, AUTH_PENDING } from '../../react/private/authStatusTypes';

const getJobsDBProps = ({ country, domainUrl, authenticationStatus, currentPage, messages }) => {
  const baseUrl = `https://${country}.${domainUrl}`;

  const iconRenderer = icon => {
    return <Icon type={icon} lineHeight='conventional' size='small' className={styles.menuIcon} />;
  };

  let icon = iconRenderer('careerInsight');

  if (country === 'th') {
    icon = iconRenderer('education');
  } else if (country === 'id') {
    icon = iconRenderer('calendar');
  }

  const profileItems =
  [{
    title: messages['header.myJobsDBResumesAndDocumentsTitle'],
    ItemIcon: iconRenderer('resume'),
    url: baseUrl + messages['header.myJobsDBResumesAndDocumentsUrl']
  },
  {
    title: messages['header.myJobsDBApplicationHistoryListTitle'],
    ItemIcon: iconRenderer('applicationHistory'),
    url: baseUrl + messages['header.myJobsDBApplicationHistoryListUrl']
  },
  {
    title: messages['header.myJobsDBJobAlertsListTitle'],
    ItemIcon: iconRenderer('jobAlert'),
    url: baseUrl + messages['header.myJobsDBJobAlertsListUrl']
  },
  {
    title: messages['header.myJobsDBSaveJobsTitle'],
    ItemIcon: iconRenderer('bookmark'),
    url: messages['header.myJobsDBSaveJobsUrl']
  }];

  const downloadItems = [
    {
      title: messages['header.playStore'],
      ItemIcon: iconRenderer('android'),
      url: messages['header.playStoreUrl']
    },
    {
      title: messages['header.appStore'],
      ItemIcon: iconRenderer('apple'),
      url: messages['header.appStoreUrl']
    }
  ];

  const links = [
    authenticationStatus === AUTHENTICATED ?
      ({ title: messages['header.myJobsDBTitle'], url: baseUrl + messages['header.myJobsDBUrlLoggedIn'], ItemIcon: iconRenderer('myJobsDB'), children: profileItems, automationId: 'myjobsDB' }) :
      ({ title: messages['header.myJobsDBTitle'], url: baseUrl + messages['header.myJobsDBUrl'], ItemIcon: iconRenderer('myJobsDB'), automationId: 'myjobsDB' }),
    { title: messages['header.resourcesTitle'], url: baseUrl + messages['header.resourcesUrl'], ItemIcon: iconRenderer('resources'), automationId: 'resources' },
    { title: messages['header.careerInsightsTitle'], url: baseUrl + messages['header.careerInsightsUrl'], ItemIcon: icon, automationId: 'career-insights' },
    { title: messages['header.downloadApp'], children: downloadItems, ItemIcon: iconRenderer('appDownload'), automationId: 'download-app', hideInDesktop: true }
  ];

  const userAccMenuItems = [
    { title: messages['header.profileTitle'], url: baseUrl + messages['header.profileUrl'], ItemIcon: ProfileIcon, EnableIcon: true },
    { title: messages['header.invitationTitle'], url: baseUrl + messages['header.invitationUrl'], ItemIcon: JobInvitationIcon, EnableIcon: true },
    { title: messages['header.logoutTitle'], url: baseUrl + messages['header.logoutUrl'], ItemIcon: JobFunctionIcon, EnableIcon: false }
  ];

  const locales = messages['header.countryList'];
  locales.map(locale => {
    locale.url = locale.language === 'id' || locale.language === 'th' ? `https://${locale.country}.${domainUrl}/${locale.country}/${locale.language}` : `https://${locale.country}.${domainUrl}/${locale.country}`;
    locale.url = (currentPage !== null && typeof(currentPage) !== 'undefined') && (locale.country === 'id' || locale.country === 'th') ? `${locale.url}/${currentPage}` : locale.url;
  });

  return {
    links,
    messages,
    userAccMenuItems,
    baseUrl,
    locales,
    profileItems
  };
};

const Header = ({ country = 'hk', language = 'en', activeTab, loginAvailable = false, selectCountry = true, authenticationStatus = UNAUTHENTICATED, userName, domainUrl, currentPage, ...restProps }) => {
  const messages = getLocalization({ country, language });
  return (

    <GlobalHeader
      LogoComponent={Logo}
      activeTab={activeTab}
      loginAvailable={loginAvailable}
      {...getJobsDBProps({ country, domainUrl, authenticationStatus, currentPage, messages })}
      brandStyles={styles}
      country={country}
      language={language}
      employerSite={(authenticationStatus === UNAUTHENTICATED) || loginAvailable}
      selectCountry={selectCountry}
      userName={userName}
      authenticationStatus={authenticationStatus}
      homeUrl={messages['header.homeUrl']}
      {...restProps}
    />
  );
};

Header.propTypes = {
  country: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired,
  activeTab: PropTypes.string,
  loginAvailable: PropTypes.bool,
  selectCountry: PropTypes.bool,
  authenticationStatus: PropTypes.oneOf([
    AUTHENTICATED,
    UNAUTHENTICATED,
    AUTH_PENDING
  ]),
  userName: PropTypes.string,
  domainUrl: PropTypes.string,
  url: PropTypes.string,
  currentPage: PropTypes.string
};

export default Header;
