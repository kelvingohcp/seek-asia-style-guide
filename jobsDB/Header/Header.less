@import (reference) "~seek-asia-style-guide/theme";

.primaryNavLink,
.primaryNavLink:visited {
  color: @dbBrand;
}

.primaryNavLink:hover {
  color: darken(@dbBrand, 10%);
}

.activeActionTrayIcon {
  color: @dbBrand;
}

.menuIcon {
  color: @dbBrand;
  fill: currentColor;
}
