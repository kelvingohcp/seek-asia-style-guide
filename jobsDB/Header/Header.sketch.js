import React from 'react';
import Header from './Header';

export const blockSymbols = {
  'JobsDB/Header/Logged out': <Header country="hk" language="en" user={{}} />,
  'JobsDB/Header/Logged in': (
    <Header country="hk" language="en" user={{ candidate: { id: 1, username: 'Rey' } }} />
  ),
  'JobsDB/Header without country and language': (
    <Header user={{ candidate: { id: 1, username: 'Rey' } }} />
  )
};
