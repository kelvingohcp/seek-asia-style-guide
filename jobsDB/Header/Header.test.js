import React from 'react';
import { shallow } from 'enzyme';

import Header from './Header';

describe('JobsDB Header', () => {
  const defaultProps = {
    country: 'hk',
    language: 'en',
    activeTab: 'ACTIVE_TAB_HOME'
  };

  it('renders with default props', () => {
    const wrapper = shallow(<Header {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
