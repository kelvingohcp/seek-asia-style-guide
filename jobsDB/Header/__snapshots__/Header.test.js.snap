// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsDB Header renders with default props 1`] = `
<Header
  LogoComponent={[Function]}
  activeTab="ACTIVE_TAB_HOME"
  authenticationStatus="unauthenticated"
  baseUrl="https://hk.jobsdb.com"
  brandStyles={null}
  country="hk"
  employerSite={true}
  homeUrl="/hk"
  language="en"
  linkRenderer={[Function]}
  links={
    Array [
      Object {
        "ItemIcon": [Function],
        "title": "My jobsDB",
        "url": "https://hk.jobsdb.com/hk/en/MyJobsDB",
      },
      Object {
        "ItemIcon": [Function],
        "title": "Resources",
        "url": "https://hk.jobsdb.com/en-hk/career-recruitment-advice/",
      },
      Object {
        "ItemIcon": [Function],
        "title": "Career Insights",
        "url": "https://hk.jobsdb.com/hk/en/career-insights",
      },
    ]
  }
  locales={
    Array [
      Object {
        "ItemIcon": [Function],
        "country": "hk",
        "language": "en",
        "languageDisplay": "English",
        "title": "Hong Kong (English)",
        "url": "https://hk.jobsdb.com/hk",
      },
      Object {
        "ItemIcon": [Function],
        "country": "id",
        "language": "en",
        "languageDisplay": "English",
        "title": "Indonesia (English)",
        "url": "https://id.jobsdb.com/id",
      },
      Object {
        "ItemIcon": [Function],
        "country": "id",
        "language": "id",
        "languageDisplay": "Bahasa",
        "title": "Indonesia (Bahasa)",
        "url": "https://id.jobsdb.com/id/id",
      },
      Object {
        "ItemIcon": [Function],
        "country": "sg",
        "language": "en",
        "languageDisplay": "English",
        "title": "Singapore (English)",
        "url": "https://sg.jobsdb.com/sg",
      },
      Object {
        "ItemIcon": [Function],
        "country": "th",
        "language": "en",
        "languageDisplay": "English",
        "title": "Thailand (English)",
        "url": "https://th.jobsdb.com/th",
      },
      Object {
        "ItemIcon": [Function],
        "country": "th",
        "language": "th",
        "languageDisplay": "ภาษาไทย",
        "title": "ไทย (ภาษาไทย)",
        "url": "https://th.jobsdb.com/th/th",
      },
    ]
  }
  loginAvailable={false}
  messages={
    Object {
      "header.careerInsightsTitle": "Career Insights",
      "header.careerInsightsUrl": "/hk/en/career-insights",
      "header.countryList": Array [
        Object {
          "ItemIcon": [Function],
          "country": "hk",
          "language": "en",
          "languageDisplay": "English",
          "title": "Hong Kong (English)",
          "url": "https://hk.jobsdb.com/hk",
        },
        Object {
          "ItemIcon": [Function],
          "country": "id",
          "language": "en",
          "languageDisplay": "English",
          "title": "Indonesia (English)",
          "url": "https://id.jobsdb.com/id",
        },
        Object {
          "ItemIcon": [Function],
          "country": "id",
          "language": "id",
          "languageDisplay": "Bahasa",
          "title": "Indonesia (Bahasa)",
          "url": "https://id.jobsdb.com/id/id",
        },
        Object {
          "ItemIcon": [Function],
          "country": "sg",
          "language": "en",
          "languageDisplay": "English",
          "title": "Singapore (English)",
          "url": "https://sg.jobsdb.com/sg",
        },
        Object {
          "ItemIcon": [Function],
          "country": "th",
          "language": "en",
          "languageDisplay": "English",
          "title": "Thailand (English)",
          "url": "https://th.jobsdb.com/th",
        },
        Object {
          "ItemIcon": [Function],
          "country": "th",
          "language": "th",
          "languageDisplay": "ภาษาไทย",
          "title": "ไทย (ภาษาไทย)",
          "url": "https://th.jobsdb.com/th/th",
        },
      ],
      "header.employerSiteTitle": "For Employers",
      "header.employerSiteUrl": "/hk/en/home/employerindex",
      "header.homeTitle": "Home",
      "header.homeUrl": "/hk",
      "header.homeUrlText": "JOBSDB",
      "header.invitationTitle": "Job invitations",
      "header.invitationUrl": "/hk/en/Invitations/InvitationList",
      "header.loginTitle": "Log in",
      "header.loginUrl": "/hk/en/login/jobseekerlogin?from=header",
      "header.logoutTitle": "Log out",
      "header.logoutUrl": "/hk/en/login/jobseekerlogoff",
      "header.mobileLoginUrl": "https://m.jobsdb.com/en-hk/login.do?nextPage=jobsForYou.do%3F",
      "header.mobileLogoutUrl": "https://m.jobsdb.com/en-hk/logout.do",
      "header.myJobsDBApplicationHistoryListTitle": "Application History",
      "header.myJobsDBApplicationHistoryListUrl": "/hk/en/MyJobsDB/ApplictationHistoryList",
      "header.myJobsDBJobAlertsListTitle": "Job Alerts",
      "header.myJobsDBJobAlertsListUrl": "/hk/en/MyJobsDB/JobAlertsList",
      "header.myJobsDBResumesAndDocumentsTitle": "Resumes and Documents",
      "header.myJobsDBResumesAndDocumentsUrl": "/hk/en/MyJobsDB/resumes-and-documents",
      "header.myJobsDBSaveJobsTitle": "Save Jobs",
      "header.myJobsDBSaveJobsUrl": "https://m.jobsdb.com/en-hk/saveJobs.do",
      "header.myJobsDBTitle": "My jobsDB",
      "header.myJobsDBUrl": "/hk/en/MyJobsDB",
      "header.myJobsDBUrlLoggedIn": "/hk/en/MyJobsDB",
      "header.profileTitle": "View profile",
      "header.profileUrl": "/hk/en/MyJobsDB/profile",
      "header.resourcesTitle": "Resources",
      "header.resourcesUrl": "/en-hk/career-recruitment-advice/",
      "header.searchUrl": "/HK/EN/Search/FindJobs?KeyOpt=COMPLEX&JSRV=1&RLRSF=1&JobCat=1&JSSRC=HPSS",
      "header.seperatorOr": "or",
      "header.signupTitle": "Sign up",
      "header.signupUrl": "/hk/en/jobseeker/registration",
      "menu.backToMenu": "Back to menu",
      "menu.countryAndLanguage": "Country & Language",
      "menu.jobSeekerHeader": "JOB SEEKER",
      "menu.jobSeekerProfile": "PROFILE",
      "menu.more": "More",
      "menu.moreHeader": "MORE",
      "menu.settingsHeader": "SETTINGS",
    }
  }
  profileItems={
    Array [
      Object {
        "ItemIcon": [Function],
        "title": "Resumes and Documents",
        "url": "https://hk.jobsdb.com/hk/en/MyJobsDB/resumes-and-documents",
      },
      Object {
        "ItemIcon": [Function],
        "title": "Application History",
        "url": "https://hk.jobsdb.com/hk/en/MyJobsDB/ApplictationHistoryList",
      },
      Object {
        "ItemIcon": [Function],
        "title": "Job Alerts",
        "url": "https://hk.jobsdb.com/hk/en/MyJobsDB/JobAlertsList",
      },
      Object {
        "ItemIcon": [Function],
        "title": "Save Jobs",
        "url": "https://m.jobsdb.com/en-hk/saveJobs.do",
      },
    ]
  }
  selectCountry={true}
  userAccMenuItems={
    Array [
      Object {
        "EnableIcon": true,
        "ItemIcon": [Function],
        "title": "View profile",
        "url": "https://hk.jobsdb.com/hk/en/MyJobsDB/profile",
      },
      Object {
        "EnableIcon": true,
        "ItemIcon": [Function],
        "title": "Job invitations",
        "url": "https://hk.jobsdb.com/hk/en/Invitations/InvitationList",
      },
      Object {
        "EnableIcon": false,
        "ItemIcon": [Function],
        "title": "Log out",
        "url": "https://hk.jobsdb.com/hk/en/login/jobseekerlogoff",
      },
    ]
  }
/>
`;
